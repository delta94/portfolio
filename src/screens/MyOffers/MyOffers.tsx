import clsx from 'clsx';
import React, { useContext } from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { Element } from 'react-scroll';
import { InformationsContext } from '../../App';
import { OffersContainer, VanishAnimation } from '../../components';
import { CustomTheme } from '../../theme';
import { Informations } from '../../utils/informations';

export interface MyOffersProps {}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  header: {
    color: theme.colors.white,
    marginLeft: theme.spacing(3),
    display: 'flex',
    justifyContent: 'center',
    cursor: 'pointer',
    marginBottom: theme.spacing(0.5),
  },
  mainContainer: {
    marginTop: theme.spacing(1.5),
  },
  mainOfferContainer: {
    display: 'flex',
    paddingLeft: theme.spacing(1),
    paddingRight: theme.spacing(1),
    justifyContent: 'center',
  },
  offerContainer: {
    display: 'flex',
    width: '93%',
    justifyContent: 'space-between',
  },
  offerContainerMobile: {
    // paddingRight: theme.spacing(2),
    // paddingLeft: theme.spacing(2),
    flexDirection: 'column',
  },
  backEndContainer: {
    marginLeft: theme.spacing(1),
  },
  backEndContainerMobile: {
    marginLeft: 0,
  },
  frontEndContainer: {
    marginRight: theme.spacing(1),
  },
  frontEndContainerMobile: {
    marginRight: 0,
  },
}));

const MyOffers: React.FC<MyOffersProps> = () => {
  const classes = useStyles();
  const context: Informations | undefined = useContext<Informations | undefined>(InformationsContext);

  const isMobile: boolean = useMediaQuery({ maxWidth: 980 });
  return (
    <Element className={classes.mainContainer} name='My Offers'>
      <div className={classes.header}>
        <VanishAnimation words={['My Offers']} />
      </div>
      <div className={classes.mainOfferContainer}>
        {context !== undefined && (
          <div className={clsx([classes.offerContainer, isMobile && classes.offerContainerMobile])}>
            <OffersContainer
              title='Frontend'
              offerings={context?.myOffers.frontEnd.offerings}
              className={clsx([classes.frontEndContainer, isMobile && classes.frontEndContainerMobile])}
            />
            <OffersContainer
              title='Backend'
              offerings={context?.myOffers.backEnd.offerings}
              className={clsx([classes.backEndContainer, isMobile && classes.backEndContainerMobile])}
            />
          </div>
        )}
      </div>
    </Element>
  );
};

export { MyOffers };
