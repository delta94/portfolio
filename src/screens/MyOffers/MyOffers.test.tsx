import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { MyOffers } from './MyOffers';

test('MyOffers renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <MyOffers />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
