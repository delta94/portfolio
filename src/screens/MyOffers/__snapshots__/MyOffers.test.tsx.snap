// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOffers renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-2"
    name="My Offers"
  >
    <div
      class="header-0-2-1"
    >
      <div
        style="opacity: 0; transform: translate3d(0,20px,0);"
      >
        <h2
          class="h2Text-0-2-10"
        >
          My Offers
        </h2>
      </div>
    </div>
    <div
      class="mainOfferContainer-0-2-3"
    />
  </div>
</DocumentFragment>
`;
